flytekit/core/container_task.py

```diff
@@ -93,8 +99,175 @@ def __init__(
+            k = self._get_key_from_cmd(cmd)
+            if k:
+                if type(native_inputs[k]) in [FlyteFile, FlyteDirectory]:
+                    local_flyte_file_or_dir_path = str(native_inputs[k])
+                    remote_flyte_file_or_dir_path = os.path.join(self._input_data_dir, k.replace(".", "/"))  # type: ignore
+                    volume_bindings[local_flyte_file_or_dir_path] = {
+                        "bind": remote_flyte_file_or_dir_path,
+                        "mode": "rw",
+                    }
+                    commands.append(remote_flyte_file_or_dir_path)
+                else:
+                    commands.append(str(native_inputs[k]))
+            else:
+                commands.append(cmd)
+
+        client = docker.from_env()
+        # If the specified image is not found locally, pull it from the registry
+        if client.images.list(filters={"reference": self._image}) == []:
+            logger.info(f"Pulling image:{self._image} for container task:{self.name}")
+            client.images.pull(self._image)
+
+        container = client.containers.run(
+            self._image,
+            command=commands,
+            remove=True,
+            volumes=volume_bindings,
+            detach=True,
+        )
+        # Wait for the container to finish the task
+        # TODO: Add a 'timeout' parameter to control the max wait time for the container to finish the task.
+        container.wait()
+
+        output_dict = {}
+        if self._outputs:
+            for k, output_type in self._outputs.items():
+                if output_type in [FlyteFile, FlyteDirectory]:
+                    output_dict[k] = output_type(path=os.path.join(output_directory, k))
+                else:
+                    with open(os.path.join(output_directory, k), "r") as f:
+                        output_val = f.read()
+
+                    # bool('False') and bool('false') are True, so we need to handle this case
+                    if output_type == bool:
+                        output_dict[k] = False if output_val.lower() == "false" else True
+                    elif output_type == datetime.datetime:
+                        output_dict[k] = datetime.datetime.fromisoformat(output_val)  # type: ignore
+                    elif output_type == datetime.timedelta:
+                        output_dict[k] = self._string_to_timedelta(output_val)
+                    else:
+                        output_dict[k] = output_type(output_val)
+
+        outputs_literal_map = TypeEngine.dict_to_literal_map(ctx, output_dict)
+        outputs_literals = outputs_literal_map.literals
+        output_names = list(self.interface.outputs.keys())
+        # Tasks that don't return anything still return a VoidPromise
+        if len(output_names) == 0:
+            return VoidPromise(self.name)
+        vals = [Promise(var, outputs_literals[var]) for var in output_names]
+        return create_task_output(vals, self.python_interface)
```

Comment:
We should override `execute()` and return output_dict here. You don't need to create promises manually.

Reply:
nice advice, thank you!!!